safekeeper/src/http/routes.rs

```diff
@@ -206,6 +204,37 @@ async fn timeline_pull_handler(mut request: Request<Body>) -> Result<Response<Bo
     json_response(StatusCode::OK, resp)
 }
 
+/// Stream tar archive with the timeline.
+async fn timeline_basebackup_handler(request: Request<Body>) -> Result<Response<Body>, ApiError> {
+    let ttid = TenantTimelineId::new(
+        parse_request_param(&request, "tenant_id")?,
+        parse_request_param(&request, "timeline_id")?,
+    );
+    check_permission(&request, Some(ttid.tenant_id))?;
+
+    let tli = GlobalTimelines::get(ttid).map_err(ApiError::from)?;
```

Comment:
See comment in `stream_basebackup_guts`. The basic question here is what we are going to do when timeline is evicted? Ideally eventually we might have coordinated eviction so that evicted state won't be dependent on the safekeeper. But so far we don't plan to do this, and it means that the `/snapshot` should work with evicted timeline as well. Seems like the easiest way to do this is to (under lock or something preventing de-eviction) 1) copy s3 partial segment under different sk id 2) stream only cfile in the snapshot. Then we should have FullAccessTimeline here, it will be returned by start_snapshot if timeline is not evicted.

Reply:
In the first version there should be the most simple logic – all code that excepts to read files from disk should use `FullAccessTimeline`. It means that WAL will be automatically downloaded if timeline is in evicted state, and the state will be automatically updated to non-evicted.

To do that, you just need to add upgrade after `GlobalTimelines::get(ttid)`: 
```
let tli = tli
        .full_access_guard()
        .await
        .map_err(ApiError::InternalServerError)?;
```

Reply:
Ok, switched to FullAccessTimeline everywhere (though later start_snapshot should be in Timeline again).